src/common/set.c

```diff
@@ -489,9 +494,17 @@ util_poolset_parse(const char *path, int fd, struct pool_set **setp)
 		return -1;
 	}
 
+	fd = dup(fd);
+	if (fd < 0) {
+		ERR("!dup");
+		return -1;
+	}
+
 	/* associate a stream with the file descriptor */
-	if ((fs = fdopen(dup(fd), "r")) == NULL) {
+	if ((fs = fdopen(fd, "r")) == NULL) {
```

Comment:
When dup works and fdopen fails we leak duplicated file descriptor, so it's necessary.